### DISCUSSION TITLE
Images don't load instantly anymore

### DISCUSSION THREAD

**Discussion by HunterPfannenstiel** [vercel/community#857] [Help] [answered]
I heard that there was a next/future/image component that had some improvements so I switched over to it yesterday. Once I switched over, I could tell that the images on pages created with getStaticProps loaded instantly every single time and it felt super smooth (this was the first time they ever loaded instantly). Today I pushed a new commit where I didn't touch anything with the images and they went back to loading slowly. I then compared the two deployments, and the images still loaded instantly with the old commit.

Slow Image Load (when selecting an item and going to it's page) - https://daylight-website-eight.vercel.app/menu
Fast Image Load (committed just yesterday) - https://daylight-website-h0duu5euz-ohshews.vercel.app/menu

What could the cause of this be? I did modify some code, but nothing to do with the images. I didn't update my Next version today. Is there a chance the new code could have affected this? (I've been working on this site for about 2 months and I have never had my images load instantly like yesterday).

**Comment by amyegan** [comment, marked as answer]
_Solution: https://github.com/vercel/community/discussions/857#discussioncomment-3704728_

---
Hi @ohShews! It is possible that new code or config changes could have affected your images, but I can't say for sure without seeing it. Is your repo public, or can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Reply by HunterPfannenstiel** [reply]
Here is the link to the repo: https://github.com/ohShews/Daylight-Website
The commit named 'Quality Fixes' on Sep. 12 is the last commit where the images load instantly. And the 'Nearly complete store front' is the commit where the images stopped loading instantly. 

When looking at the two sites, I noticed that when clicking a menu item and going to its page, the site where the images loaded instantly did not 're-fetch' the image (when looking at the 'network' tab in the dev tools, no new image I served to the client) whereas the slower site did 're-fetch' the images.

'./components/donut-preview/donut-item.js' is where the image component is rendered.

*Note: I did notice yesterday while in dev mode it wanted me to add the 'sizes' prop to the images to 'improve performance so I did, but I don't recall seeing the warning two days ago.

**Reply by amyegan** [reply]
Thanks for sharing this! The only thing that stands out to me is the use of `rem` in the [`sizes` prop](https://nextjs.org/docs/api-reference/next/future/image#sizes). But I'm not sure that would cause the kind of delay we're talking about. I'm sharing this with the Next.js engineers to get their advice.

**Reply by amyegan** [reply]
Update: from what I can tell, the "instant" image loading is a bit of an illusion. The team is suggesting that you try adding a loading indicator to the donut details modal and use `onLoadingComplete` to show it after the content is all ready. It seems like something else in the older version may have been causing enough of a delay on the modal to make the image appear to load instantly.

**Reply by HunterPfannenstiel** [reply]
Will 'onLoadingComplete' allow me to only visit the page once the image is fully loaded? From my understanding, 'onLoadingComplete' calls a function with the image element as a parameter when the image is loaded. How could I delay the page load until it is ready?

**Reply by amyegan** [reply]
[onLoadingComplete](https://nextjs.org/docs/api-reference/next/image#onloadingcomplete) is a callback function that you could use to signal the modal component to change its status from "loading" to "ready". You would need to implement the logic and appearance in whatever way makes sense to you for the website you're building.

**Reply by HunterPfannenstiel** [reply]
I think I finally realized the reason why the images loaded instantly on the one site. It was because none of the images had the 'sizes' prop on the instant load site so the site only needed to fetch the images once (and this fetching would happen when you went to the menu page) because it would then be the same exact image when you clicked on an item.

When having two images each with different 'sizes' props, the site wants to refetch the images because it wants a different size, even though the source is the exact same. 

When using the same 'sizes' props for each image component (with the same exact source), I was able to achieve the instant image load.
